elastic/logstash
# Fixes centos-7+ package

## What does this PR do?
Fix the CentOS 7 package installation when shipping both an sys-v init script and systemd unit. This is so by modernizing the sys-v init script to rely on 'functions'.



## Why is it important/What is the impact to the user?

Users installing on centos-7 and using 'service' script to manage services, would have failed to use the systemd unit, and instead, would have used the sys-v init script. This is due to the fact that when not using the 'functions' script, RHEL based systems rely first on sys-v initscript vs allowing it to fallback to systemd units if both were shipped.

Ubuntu doesn't have the same problem provided that the OS itself first relies on systemd units to later fallback to sys-v if not present.



## Checklist




## Author's Checklist



## How to test this PR locally
1. Create an RPM and install it.
2. Run 'service logstash start'
3. Check that the output is from systemd.
